tests/quic.c

```diff
@@ -0,0 +1,1390 @@
+
+    /* Should have received a session ticket, save the session */
+    AssertTrue(tclient.ticket_len > 0);
+    AssertNotNull(session = wolfSSL_get1_session(tclient.ssl));
+    QuicTestContext_free(&tserver);
+    QuicTestContext_free(&tclient);
+
+    /* Do a Session resumption with the ticket */
+    QuicTestContext_init(&tserver, ctx_s, "server", verbose);
+    QuicTestContext_init(&tclient, ctx_c, "client_resume", verbose);
+    AssertIntEQ(wolfSSL_set_session(tclient.ssl, session), WOLFSSL_SUCCESS);
+    /* let them talk */
+    QuicConversation_init(&conv, &tclient, &tserver);
+    QuicConversation_do(&conv);
+    /* this is what should happen. Look Ma, no certificate! */
+    AssertStrEQ(conv.rec_log, "ClientHello:ServerHello:EncryptedExtension:Finished:Finished:SessionTicket");
+
+    QuicTestContext_free(&tclient);
+    QuicTestContext_free(&tserver);
+
+    wolfSSL_SESSION_free(session);
+    wolfSSL_CTX_free(ctx_c);
+    wolfSSL_CTX_free(ctx_s);
+
+    printf("    test_quic_resumption: %s\n", (ret == 0)? passed : failed);
+    return ret;
+}
+
+#ifdef WOLFSSL_EARLY_DATA
+static int test_quic_early_data(int verbose) {
+    WOLFSSL_CTX *ctx_c, *ctx_s;
+    int ret = 0;
+    QuicTestContext tclient, tserver;
+    QuicConversation conv;
+    const byte early_data[] = "Nulla dies sine linea!";
+    size_t ed_written;
+    WOLFSSL_SESSION *session;
+
+    AssertNotNull(ctx_c = wolfSSL_CTX_new(wolfTLSv1_3_client_method()));
+    wolfSSL_CTX_UseSessionTicket(ctx_c);
+
+    AssertNotNull(ctx_s = wolfSSL_CTX_new(wolfTLSv1_3_server_method()));
+    AssertTrue(wolfSSL_CTX_use_certificate_file(ctx_s, svrCertFile, WOLFSSL_FILETYPE_PEM));
+    AssertTrue(wolfSSL_CTX_use_PrivateKey_file(ctx_s, svrKeyFile, WOLFSSL_FILETYPE_PEM));
+
+    /* setup ssls */
+    QuicTestContext_init(&tclient, ctx_c, "client", verbose);
+    QuicTestContext_init(&tserver, ctx_s, "server", verbose);
+    wolfSSL_set_quic_early_data_enabled(tserver.ssl, 1);
+    /* QUIC only allows 0xffffffff or 0x0 as values */
+    AssertIntEQ(wolfSSL_get_max_early_data(tserver.ssl), UINT32_MAX);
+
+    QuicConversation_init(&conv, &tclient, &tserver);
+    /* run till end */
+    QuicConversation_do(&conv);
+    /* what have we seen? */
+    AssertStrEQ(conv.rec_log, "ClientHello:ServerHello:EncryptedExtension:Certificate:CertificateVerify:Finished:Finished:SessionTicket");
+
+    /* Should have received a session ticket, save the session */
+    AssertTrue(tclient.ticket_len > 0);
+    AssertNotNull(session = wolfSSL_get1_session(tclient.ssl));
+    QuicTestContext_free(&tclient);
+    QuicTestContext_free(&tserver);
+
+    /* QUIC requires 0 or 0xffffffff as only allowed values.
+     * Since we enabled early data in the server that created the session,
+     * we need to see it here. */
+    unsigned int max_early_sz = wolfSSL_SESSION_get_max_early_data(session);
```

Comment:
Please have all variable declarations at top of function or brace:
```tests/quic.c:1228:5: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
 1228 |     unsigned int max_early_sz = wolfSSL_SESSION_get_max_early_data(session);
      |     ^~~~~~~~
```

Reply:
Changed.